### ISSUE TITLE
Domain groups page: Delete a domain group

### ISSUE THREAD

**Issue by vickyszuchin** [cisagov/manage.get.gov#2434]
### Issue description

Domain groups are meant to act like a label on a domain. These will be set by admin users to "group" domains together to help with their own internal organization. For instance, some users may group domains that have a particular manager or who expire within the same month. Just like in github where we can filter by a label, the goal is to let users filter by any set of domain groups (labels).

To manage these domain groups, the domain group page should allow the user to **delete** a domain group that is not needed anymore, or **update** an existing domain group.

### Acceptance criteria
- [ ] The user can delete a domain group (delete button)
  - [ ] provide confirmation modal to confirm user's wish to delete the domain group (content TBD)
  - [ ] Display a modal with (content TBD) to confirm that your changes will be lost if you navigate away from editing
- [ ] Implementation conforms to specifications in figma design (TBD)

Implementation considerations:
- Allow editing of one domain group at time
- You can not create new domain group and edit another domain group at the same time
- If the user navigates away from editing (by clicking the new group button, or filtering the table for instance), present user with modal to confirm that their changes will be lost. 
  - The intent is to ensure that only one "action" can occur at a time. For instance, you _cannot_ add a new domain group and also edit a record

### Additional context

For implementation details, see **(figma TBD)**

Refer [to this document](https://docs.google.com/document/d/1jG_HmilrZ8cTm5WBeSQBwUmXqzzoNT8chVdtxujxYbk/edit#bookmark=kix.bht38piwufep) for details regarding edit/delete permissions. Org admins and readonly org admins can make alterations to this page.
See issue #2433

### Links to other issues

⛔ Blocked by:
- https://github.com/cisagov/manage.get.gov/issues/2396
- https://github.com/cisagov/manage.get.gov/issues/2435

🔄 Related to:
- https://github.com/cisagov/manage.get.gov/issues/2433

**Comment by katypies** [comment]
Closing this. We don't currently have a sense of whether domain groups will be needed. We can re-open or create a new issue if this comes up in our user interviews or usability testing. I suspect that suborgs may solve many of the intended use-cases here, but it's worth waiting to determine if this is a useful addition in the future.
